cypress/e2e/cypress/with-ladybug-ff-test-webapp/filtering.cy.ts

```diff
@@ -0,0 +1,102 @@
+const NUM_COLUMNS_WHITE_BOX_VIEW = 10
+
+describe('Tests with views and filtering', () => {
+  before(() => {
+    cy.apiDeleteAll('FileDebugStorage')
+    cy.apiDeleteAll('Test')
+    cy.visit('')
+    cy.runInTestAPipeline('Example1a', 'Adapter1a', 'xxx')
+    cy.runInTestAPipeline('Example1b', 'Adapter1b', 'xxx')
+    cy.runInTestAPipeline('Example1c', 'Adapter1c', 'xxxyyy')
+    cy.runInTestAPipeline('UseToStreamPipe', 'UseToStreamPipeChar', 'yyy')
+    cy.runInTestAPipeline('UseToStreamPipe', 'UseToStreamPipeBin', 'yyy')
+    cy.getNumLadybugReports().should('equal', 5)
+  })
+
+  interface ColumnAndName {
+    readonly name: string
+    readonly colNr: number
+    readonly labelFilterPanel: string
+  }
+
+  const columnAndNameCombinations: ColumnAndName[] = [
+    { name: 'Storage Id', colNr: 1, labelFilterPanel: 'Storageid' },
+    { name: 'End time', colNr: 2, labelFilterPanel: 'Endtime' },
+    { name: 'Duration', colNr: 3, labelFilterPanel: 'Duration' },
+    { name: 'Name', colNr: 4, labelFilterPanel: 'Name' },
+    { name: 'Correlation Id', colNr: 5, labelFilterPanel: 'Correlationid' },
+    { name: 'Status', colNr: 6, labelFilterPanel: 'Status' },
+    { name: 'Checkpoints', colNr: 7, labelFilterPanel: 'Numberofcheckpoints' },
+    { name: 'Memory', colNr: 8, labelFilterPanel: 'Estimatedmemoryusage' },
+    { name: 'Size', colNr: 9, labelFilterPanel: 'Storagesize' },
+    { name: 'Input', colNr: 10, labelFilterPanel: 'Input' }
+  ]
+
+  const testedColumnAndNameCombinations = columnAndNameCombinations.filter((testCase) => testCase.name !== 'Status')
+
+  for (const testCase of testedColumnAndNameCombinations) {
+    it(`Filter on field ${testCase.name}, expected at column ${testCase.colNr}`, () => {
```

Comment:
Now you're looping over the list twice. It's better to use the if statement in the for loop to reduce the time complexity like so:
```suggestion
  for (const testCase of testedColumnAndNameCombinations) {
    if (testCase.name !== 'Status') {
      it(`Filter on field ${testCase.name}, expected at column ${testCase.colNr}`, () => {
```

Reply:
Ik focus niet al te veel op performance, maar vooral op leesbaarheid. Verschillen tussen bijvoorbeeld O(n^2) of O(n) zijn belangrijk, verschillen tussen O(n) of 2 * O(n) vind ik minder belangrijk.